test/Groups/matrixgroups.jl

```diff
@@ -0,0 +1,421 @@
+@testset "Oscar-GAP relationship" begin
+   F = GF(29,1)[1]
+   z = F(2)
+   G = GL(3,F)
+   @test G.X isa GapObj
+   @test isdefined(G,:X)
+   @test isdefined(G, :ring_iso)
+   @test isdefined(G, :mat_iso)
+   Z = G.ring_iso(z)
+   @test GAP.Globals.IN(Z,G.ring_iso.codomain)
+   @test G.ring_iso(Z)==z
+   @test G.ring_iso.domain==F
+   @test GAP.Globals.IsField(G.ring_iso.codomain)
+   @test GAP.Globals.Size(G.ring_iso.codomain)==29
+   @test GAP.Globals.IsZero(14*Z+1)
+   @test iszero(G.ring_iso(GAP.Globals.Zero(G.ring_iso.codomain)))
+   @test GAP.Globals.IsOne(G.ring_iso(one(F)))
+   @test isone(G.ring_iso(GAP.Globals.One(G.ring_iso.codomain)))
+   
+   xo = matrix(F,3,3,[1,z,0,0,1,2*z+1,0,0,z+2])
+   xg = Vector{GapObj}(undef, 3)
+   for i in 1:3
+      xg[i] = GAP.julia_to_gap([G.ring_iso(xo[i,j]) for j in 1:3])
+   end
+   xg=GAP.julia_to_gap(xg)
```

Comment:
I recommend to avoid `GAP.julia_to_gap` and instead use `GapObj`:
```suggestion
   xg = Vector{GapObj}(undef, 3)
   for i in 1:3
      xg[i] = GapObj([G.ring_iso(xo[i,j]) for j in 1:3])
   end
   xg=GAP.julia_to_gap(xg)
```

But perhaps even better would be to use a recursive conversion (this is *untested*, so don't blindly merge this suggestion)
```suggestion
   xg=GapObj([G.ring_iso(xo[i,j]) for j in 1:3 for j in 1:3]; recursive = true)
```

Reply:
The line `xg = GapObj([[G.ring_iso(xo[i,j]) for j in 1:3] for i in 1:3]; recursive=true)` works and passes the tests. Do I leave it?